src/main/java/com/vmware/vipclient/i18n/base/cache/CacheItem.java

```diff
@@ -4,6 +4,51 @@
  */
 package com.vmware.vipclient.i18n.base.cache;
 
-public interface CacheItem {
-    public boolean isExpired();
+import com.vmware.vipclient.i18n.VIPCfg;
+
+public abstract class CacheItem {
+    private String etag;
+    private long timestamp;
+    private Long maxAgeMillis = 86400000l;
+
+    public String getEtag() {
+        return etag;
+    }
+
+    protected void setEtag(String etag) {
+        this.etag = etag;
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    protected void setTimestamp(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    public Long getMaxAgeMillis() {
```

Comment:
make this synchronized ?

Reply:
Same comment: It has been like this before, the only change in this PR is to put common code inside the common CacheItem instead of implementing twice in MessageCacheItem and FormatCacheItem. Let's make any code logic change separately.

Also, why does it have to be synchronized? The isExpired method already has a synchronized block. Other usage of geMaxAgeMillis aside from inside isExpired do not need to be synchronized.

Reply:
Hmm, let's discuss that separately. This PR is only to merge duplicate code into one.